ssz/src/test/java/tech/pegasys/teku/ssz/backing/ContainersGenerator.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.ssz.backing;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ContainersGenerator {
```

Comment:
I think I'd move this to a new module just for this generator. Then it can live in the `main` source tree for that module and we should be able to easily create a Gradle task that regenerates the classes.

I like having the generated classes checked in given they won't change much - it may be worth having a task in Gradle that tests the generated source hasn't been changed.  It would run the generator, outputting the result to a temp directory and then compare what was generated with what's in the actual source.  That way we'll avoid issues of people modifying the generated source and then those changes being lost on the next regeneration.  Just need to ensure the generated code complies with spotless or it will wind up with different formatting - hopefully that's not too hard (if it is we could potentially do the comparison completely ignoring whitespace and it would still be effective).

In terms of template placing, I like that they can be treated as actual java source code so I'd keep them in the same source tree as the generator (so that would wind up being the `main` tree in the new module).